network/src/peers/peer/outbound_handler.rs

```diff
@@ -100,16 +100,29 @@ impl Peer {
     ) -> Result<PeerResponse, NetworkError> {
         match message {
             PeerAction::Disconnect => Ok(PeerResponse::Disconnect),
-            PeerAction::Send(message) => {
+            PeerAction::Send(message, time_received) => {
                 if matches!(message, Payload::Ping(_)) {
                     self.quality.expecting_pong = true;
                     self.quality.last_ping_sent = Some(Instant::now());
                 }
+
+                let histogram = match &message {
+                    Payload::Peers(_) => Some(metrics::internal_rtt::GETPEERS),
+                    Payload::Sync(_) => Some(metrics::internal_rtt::GETSYNC),
+                    Payload::SyncBlock(_, _) => Some(metrics::internal_rtt::GETBLOCKS),
+                    Payload::MemoryPool(_) => Some(metrics::internal_rtt::GETMEMORYPOOL),
+                    _ => None,
+                };
+
                 network.write_payload(&message).await.map_err(|e| {
                     metrics::increment_counter!(metrics::outbound::ALL_FAILURES);
                     e
                 })?;
 
+                if let (Some(time_received), Some(histogram)) = (time_received, histogram) {
+                    metrics::histogram!(histogram, time_received.elapsed());
+                }
+
```

Comment:
I think it would be best to have this together with the block above, perhaps adjusted to `if let Some(histogram) = match &message`

Reply:
Agreed; consolidated 👍